### PR TITLE
[PT-BR] Translating Type Checking JavaScript Files

### PR DESCRIPTION
**Pull request by luk3skyw4lker**
Starting translation and creating draft PR

### CODE REVIEW

File: packages/documentation/copy/pt/javascript/Type Checking JavaScript Files.md [hunk 1]
```diff
@@ -0,0 +1,310 @@
+---
+title: Checando tipos de arquivos JavaScript
+layout: docs
+permalink: /pt/docs/handbook/type-checking-javascript-files.html
+oneline: Como adicionar checagem de tipos a arquivos JavaScript usando Typescript
+---
+
+Aqui estão algumas diferenças notáveis em como a checagem de tipo funciona em arquivos `.js` comparados com arquivos `.ts`.
+
+## Propriedades sao inferidas de atribuiçoes no corpo das classes
```
Comment (khaosdoctor) [inline packages/documentation/copy/pt/javascript/Type Checking JavaScript Files.md]:
```suggestion
## Propriedades são inferidas de atribuições no corpo das classes
```

File: packages/documentation/copy/pt/javascript/Type Checking JavaScript Files.md [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+
+ES2015 nao tem meios de declaração de propriedades em uma classe. Propriedades são atribuídas dinamicamente, assim como object literals.
```
Comment (khaosdoctor) [inline packages/documentation/copy/pt/javascript/Type Checking JavaScript Files.md]:
```suggestion
ES2015 não tem meios de declaração de propriedades em uma classe. Propriedades são atribuídas dinamicamente, assim como objetos literais.
```

File: packages/documentation/copy/pt/javascript/Type Checking JavaScript Files.md [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+
+Em um arquivo `.js`, o compilador infere propriedades a partir da atribuição de propriedades dentro de um corpo de uma classe.
```
Comment (khaosdoctor) [inline packages/documentation/copy/pt/javascript/Type Checking JavaScript Files.md]:
```suggestion
Em um arquivo `.js`, o compilador infere propriedades a partir da atribuição das mesmas dentro de um corpo de uma classe.
```

File: packages/documentation/copy/pt/javascript/Type Checking JavaScript Files.md [hunk 4]
```diff
@@ -0,0 +1,310 @@
...
+O tipo de uma propriedade é o tipo dado no construtor, a não ser que não seja definido lá ou o tipo no construtor for undefined ou null.
+Neste caso, o tipo é uma união dos tipos de todos os valores que estao do lado direito nessas atribuições.
```
Comment (khaosdoctor) [inline packages/documentation/copy/pt/javascript/Type Checking JavaScript Files.md]:
```suggestion
Neste caso, o tipo é uma união dos tipos de todos os valores que estão do lado direito dessas atribuições.
```

File: packages/documentation/copy/pt/javascript/Type Checking JavaScript Files.md [hunk 5]
```diff
@@ -0,0 +1,310 @@
...
+    this.constructorUnknown = "plunkbat"; // ok, construtorUnknown é uma string | undefined
+    this.methodOnly = "ok"; // ok, mas methodOnly também pode ser undefined
+  }
+  method2() {
+    this.methodOnly = true; // também ok, methodOnly's type é string | boolean | undefined
+  }
+}
+```
+
+Se propriedades nunca forem declaradas no corpo da classe, elas são consideradas desconhecidas.
```
Comment (khaosdoctor) [inline packages/documentation/copy/pt/javascript/Type Checking JavaScript Files.md]:
```suggestion
Se propriedades nunca forem declaradas no corpo da classe, elas são consideradas desconhecidas (com o tipo unknown).
```

File: packages/documentation/copy/pt/javascript/Type Checking JavaScript Files.md [hunk 6]
```diff
@@ -0,0 +1,310 @@
...
+    this.count;
+  }
+}
+
+let c = new C();
+c.prop = 0; // OK
+c.count = "string";
+```
+
+## Funções construtoras sao equivalentes a classes
```
Comment (khaosdoctor) [inline packages/documentation/copy/pt/javascript/Type Checking JavaScript Files.md]:
```suggestion
## Funções construtoras são equivalentes a classes
```

File: packages/documentation/copy/pt/javascript/Type Checking JavaScript Files.md [hunk 7]
```diff
@@ -0,0 +1,310 @@
...
+// o mesmo que `export funcion readFile`
+module.exports.readFile = function (f) {
+  return fs.readFileSync(f);
+};
+```
+
+O suporte de módulo em Javascript é muito mais flexível sinteticamente que o suporte de módulo de TypeScript.
+Muitas das combinações de atribuições e declarações são suportadas.
+
+## Classes, funções, e object literals sao namespaces
```
Comment (khaosdoctor) [inline packages/documentation/copy/pt/javascript/Type Checking JavaScript Files.md]:
```suggestion
## Classes, funções, e object literals são namespaces
```

File: packages/documentation/copy/pt/javascript/Type Checking JavaScript Files.md [hunk 8]
```diff
@@ -0,0 +1,310 @@
...
+  };
+assign.extra = 1;
+```
+
+## Objetos literais são abertos
+
+Em um arquivo `.ts`, um objeto literal que inicializa uma declaração de variável dá o seu tipo para a declaração.
+Nenhum membro novo que não foi especificado na declaração pode ser adicionado.
+Essas regras são relaxadas em um arquivo `.js`; objetos literais tem um tipo aberto (uma assinatura de índice) que permite adicionar e procurar propriedades que não foram adicionadas originalmente.
+Por exempo:
```
Comment (khaosdoctor) [inline packages/documentation/copy/pt/javascript/Type Checking JavaScript Files.md]:
```suggestion
Por exemplo:
```

File: packages/documentation/copy/pt/javascript/Type Checking JavaScript Files.md [hunk 9]
```diff
@@ -0,0 +1,310 @@
...
+
+```js twoslash
+var obj = { a: 1 };
+obj.b = 2; // Permitido
+```
+
+Objetos literais se comportam como se tivessem uma assinatura de index `[x:string]: any` que permite que sejam tratados como maps abertos ao invés de objetos fechados.
```
Comment (khaosdoctor) [inline packages/documentation/copy/pt/javascript/Type Checking JavaScript Files.md]:
```suggestion
Objetos literais se comportam como se tivessem uma assinatura de índice `[x:string]: any` que permite que sejam tratados como maps abertos ao invés de objetos fechados.
```

File: packages/documentation/copy/pt/javascript/Type Checking JavaScript Files.md [hunk 10]
```diff
@@ -0,0 +1,310 @@
...
+```
+
+## Parâmetros de funções são opcionais por padrão
+
+Uma vez que não há uma forma de especificar opcionalidade em parâmetros de funções em Javascript pre-ES2015, todas os parâmetros de funções são considerados opcionais.
+Chamadas com menos argumentos que a quantidade declarada na função são permitidas.
+
+É importante notar que é um erro chamar funções com mais argumentos do que declarados.
+
+For instance:
```
Comment (khaosdoctor) [inline packages/documentation/copy/pt/javascript/Type Checking JavaScript Files.md]:
```suggestion
Por exemplo:
```

**Review by khaosdoctor — CHANGES_REQUESTED** [review]
Some corrections

**Comment by orta** [comment]
This PR will need a rebase BTW (I can't push one)

**Comment by luk3skyw4lker** [comment]
> This PR will need a rebase BTW (I can't push one)

Could you be more clear about how the rebase should look like? I was fearing that I would have to do one, but I don't really know how or if I should really do it.

Edit: How the commits should look like**

**Review by khaosdoctor — APPROVED** [review]

**Comment by khaosdoctor** [comment]
lgtm

**Comment by khaosdoctor** [comment]
well, it went through... That's weird

**Comment by orta** [comment]
Interesting, maybe the API allows for merges even when the UI says it's not feasible - I added code to check that it's mergeable to the bot

**Pull request merged**